StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Link/LinkPaymentController.swift

```diff
@@ -0,0 +1,221 @@
+    /// - Note: Once `confirm(from:)` returns successfully, calling this method is an error, as payment/setup intents should not be reused. Until then, you may call this method as many times as is necessary.
+    /// - Parameter presentingViewController: The view controller to present the payment flow from
+    /// - Throws: Either `LinkPaymentController.Error.canceled`, meaning the customer canceled the flow, or an error describing what went wrong.
+    @MainActor
+    @_spi(LinkOnly) public func present(from presentingViewController: UIViewController) async throws -> Void {
+        let linkController: PayWithLinkViewController = try await withCheckedThrowingContinuation { [self] continuation in
+            PaymentSheet.load(clientSecret: intentSecret, configuration: configuration) { result in
+                switch result {
+                case .success(let intent, _, let isLinkEnabled):
+                    guard isLinkEnabled else {
+                        continuation.resume(throwing: LinkPaymentController.Error.unavailable)
+                        return
+                    }
+                    self.intent = intent
+                    let linkController = PayWithLinkViewController(
+                        intent: intent,
+                        configuration: self.configuration,
+                        shouldOfferApplePay: false,
+                        shouldFinishOnClose: true,
+                        shouldPayInFlow: false
+                    )
+                    continuation.resume(returning: linkController)
+                case .failure(let error):
+                    continuation.resume(throwing: error)
+                }
+            }
+        }
+
+        linkController.payWithLinkDelegate = self
+        linkController.modalPresentationStyle = UIDevice.current.userInterfaceIdiom == .pad
+            ? .formSheet
+            : .overFullScreen
+
+        defer { linkController.dismiss(animated: true) }
+        try await withCheckedThrowingContinuation { (continuation: CheckedContinuation<Void, Swift.Error>) in
+            payWithLinkContinuation = continuation
+            presentingViewController.present(linkController, animated: true)
+        }
+    }
+
+    /// Completes the Link payment or setup.
+    /// - Note: Once `completion` is called with a `.completed` result, this `LinkPaymentController` instance should no longer be used, as payment/setup intents should not be reused. Other results indicate cancellation or failure, and do not invalidate the instance.
+    /// - Parameter presentingViewController: The view controller used to present any view controllers required e.g. to authenticate the customer
+    /// - Parameter completion: Called with the result of the payment after any presented view controllers are dismissed
+    @_spi(LinkOnly) public func confirm(from presentingViewController: UIViewController, completion: @escaping (PaymentSheetResult) -> Void) -> Void {
```

Comment:
Ignore this line of code, just wondering a general question that I'm sure you've thought about but - I would imagine some merchants want to offer Link like they offer Apple Pay - as a standalone button that, when tapped, displays a modal that completes the payment in one go.  This is in contrast to the "two-step" UX in this PR.  

Does that make sense?  Maybe this PR actually allows for this already (call present -> immediately call confirm afterwards?)

Reply:
As with the private beta comment, this is complicated by the fact that this class is being built for a very small number of specific merchants rather than (at least at present) general merchant use, and those merchants specifically requested a two-step confirmation flow. The flow you've described is the standard Link flow, and how it currently works when you tap the button within PaymentSheet - so far, though, we haven't made Link available as a separate wallet button on mobile, where that behavior would be expected. Should we decide to pursue that, we would absolutely need to handle that use case (probably by leveraging `PayWithLinkController` instead, which is what PaymentSheet uses). For now, I'm opting to omit it until there's a demonstrated need for it.